src/meltano/core/project.py

```diff
@@ -210,44 +203,41 @@ def find(cls, project_root: Path | str = None, activate=True):
 
         return project
 
-    @property
-    def project_files(self):
-        """Return a singleton ProjectFiles file manager instance.
+    @cached_property
+    def project_files(self) -> ProjectFiles:
+        """Return a singleton `ProjectFiles` file manager instance.
 
         Returns:
-            ProjectFiles file manager
+            `ProjectFiles` file manager.
         """
-        if self._project_files is None:
-            self._project_files = ProjectFiles(
-                root=self.root, meltano_file_path=self.meltanofile
-            )
-        return self._project_files
+        return ProjectFiles(root=self.root, meltano_file_path=self.meltanofile)
+
+    def clear_cache(self) -> None:
+        """Clear cached project files (e.g. `meltano.yml`)."""
```

Comment:
Can you add some guidance in the docstring about in which circumstances or stages it's recommended to clear the project file cache?

Reply:
Good call. I added the following:

> This can be useful if the cached `ProjectFiles` object has been modified in-place, but not updated on-disk, and you need the on-disk version.

It's a pretty niche situation, but we have some tests that explicitly check it.